dotcom-rendering/src/components/AppsEpic.importable.tsx

```diff
@@ -0,0 +1,125 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css } from '@emotion/react';
+import type { IGetEpics__Result } from '@guardian/bridget/Acquisitions';
+import {
+	body as bodySizes,
+	from,
+	headline,
+	palette,
+	remSpace,
+	textSans,
+} from '@guardian/source-foundations';
+import { useEffect, useState } from 'react';
+import { getAcquisitionsClient, getUserClient } from '../lib/bridgetApi';
+import { EpicContent } from './EpicContent.apps';
+
+export const AppsEpic = () => {
+	const [isPremium, setIsPremium] = useState<boolean | undefined>(undefined);
+	const [maybeEpic, setMaybeEpic] =
+		useState<IGetEpics__Result['success']>(undefined);
+
+	useEffect(() => {
+		void getUserClient()
+			.isPremium()
+			.then(setIsPremium)
+			.catch(() => undefined);
```

Comment:
Because we're using the `void` above and are therefore not explicitly handling the result from this promise i think we can remove the .catch()

Reply:
Oh that's interesting! I assumed we'd get an uncaught exception error in the console if we don't add a catch, even with `void`. I did a very quick test in my browser console and it looks like `void` doesn't squash the unhandled promise rejection:

<img width="398" alt="image" src="https://github.com/guardian/dotcom-rendering/assets/705427/c7f1c363-6311-41d9-b098-9138b3d4e76f">

Reply:
😱 well, ignore me then!

Reply:
This is interesting. `typescript-eslint` suggests using `void` is a valid way to intentionally ignore the result of a promise, whether it resolves or rejects[^1]. In fact if you remove the `void` here you get the following linter error:

```
Promises must be awaited, end with a call to .catch, end with a call to .then
with a rejection handler or be explicitly marked as ignored with the `void`
operator
```

which suggests using `void` as a valid alternative to `.catch`. @abeddow91 that's what I had in mind when I made this comment: https://github.com/guardian/dotcom-rendering/pull/8834#discussion_r1330107676

We use this pattern in many places across DCR and AR, presumably as a way of marking promises that we don't mind failing occasionally and therefore for which we don't want to do anything to handle that failure.

Is the suggestion here that this may not be a good idea because we still get an error in the console, and we should be using `catch(() => undefined)` instead of `void`? If so we may want to set `ignoreVoid: false`[^1] in the linter configuration so that it warns about these cases (I think the default is `true`)?

[^1]: https://typescript-eslint.io/rules/no-floating-promises/#ignorevoid

Reply:
@sophie-macmillan I don't think this is a blocker for your PR btw, I've just unresolved this thread so it's visible.